src/validate.ts

```diff
@@ -97,23 +113,97 @@ export function validate(
         ),
       ];
     }
+    if (ts.isCallExpression(node)) {
```

Comment:
Do we validate that they don't use promise.then ?

Reply:
It would just say that Promise.then is an unsupported call. I can add more specific errors.

Reply:
In `validate.ts` or at synth time?

Reply:
```ts
    + https://functionless.org/docs/error-codes#integration-must-be-immediately-awaited-or-returned
    +
    + 96   return await func().then((x) => x);
    +                  ~~~~~~~
    + test/test-files/step-function.ts:100:15 - error Functionless(10016): Integration must be immediately awaited or returned
```

I think this is OK for now.